plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/parser/SQLSemanticProcessor.java

```diff
@@ -62,12 +62,16 @@ public class SQLSemanticProcessor {
     private static final Log log = Log.getLog(SQLSemanticProcessor.class);
 
     private static final boolean ALLOW_COMPLEX_PARSING = false;
+    private static final int PARSE_FUTURE_TIMEOUT_MS = 10000; // 8000 is the default timeout for parser itself
 
     public static Statement parseQuery(@Nullable SQLDialect dialect, @NotNull String sql) throws DBCException {
+
+
         String sqlWithoutComments = dialect == null ? sql : SQLUtils.stripComments(dialect, sql);
-        CCJSqlParser parser = new CCJSqlParser(new StringProvider(sqlWithoutComments));
+        ExecutorService executorService =  Executors.newSingleThreadExecutor();

```

Comment:
You're not terminating the executor. Judging by the docs, it will be disposed of once it becomes phantom reachable, but we can do that ourselves here. Moreover, we have a suitable `try` block just a line below :^)